test/apps/iodemo/io_demo.cc

```diff
@@ -846,10 +856,12 @@ class DemoClient : public P2pDemoCommon {
 
             timersub(&tv_curr, &tv_start, &tv_diff);
             double elapsed = tv_diff.tv_sec + (tv_diff.tv_usec * 1e-6);
-            if (elapsed > _test_opts.client_timeout * 10) {
+            if ((elapsed > _test_opts.client_timeout * 10) && (_status = OK)) {
                 LOG << "timeout waiting for " << (_num_sent - _num_completed)
                     << " replies";
                 _status = WAIT_RESPONSES_TIMEOUT;
+                /* cancel all but wait counters */
+                do_io_cancel_all();
```

Comment:
we should cancel (and close connection) only those servers which have uncompleted (e.g, timed-out) operations